dive_core/gfxr_vulkan_command_hierarchy.h

```diff
@@ -0,0 +1,61 @@
+/*
+ Copyright 2025 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+// =====================================================================================================================
+// The VulkanCommandHierarchy class parses and creates a tree of Nodes in the command buffer. The
+// primary client for this class is the Model class for the Vulkan command and argument Views in the
+// UI.
+// =====================================================================================================================
+
+#include "dive_core/command_hierarchy.h"
+
+namespace Dive
+{
+class GfxrVulkanCommandHierarchyCreator : public IEmulateCallbacks
+{
+public:
+    GfxrVulkanCommandHierarchyCreator(CommandHierarchy &command_hierarchy,
+                                      CaptureData      &capture_data);
+
+    bool CreateTrees();
+
+private:
+    enum ChildListType
```

Comment:
Thanks for making this change! can you apply the same change (use enum) to CommandHierarchyCreator? (again, you can put the change in a different PR if you feel it is too much for this PR) thanks

Reply:
Sure! Will put in a separate PR.